### ISSUE TITLE
Branch out ActivityPub library

### ISSUE THREAD

**Issue by stevenroose** [rustodon/rustodon#204]
It would be nice if all the code that implement the basic ActivityPub primitives would be accessible in a standalone library crate.

It would be very useful for developers trying to do other AP-related projects. It would also be better software design for rustodon, IMO.

**Comment by BaptisteGelez** [comment]
I don't know how ActivityPub is currently implemented, but the `activitypub` crate provides type-safe manipulation of AP objects.

**Comment by barzamin** [comment]
Definitely. I remember looking at the activitypub crate and finding a few things that I thought might be problematic for using it, but that was almost a year ago now and perhaps things have changed!

**Comment by asonix** [comment]
Feel free to let me know what sorts of things you'd like to see from the activitypub crate. Right now, it's basically just a few types and traits related to the activitypub spec that attempt to help with what can go where.

**Comment by stevenroose** [comment]
Personally probably mostly the JSON object specs. So that I can parse and build ActivityStreams and ActivityPub related objects. I don't exactly know the Pub<>Streams distinction spec-wise. Might as well be that all JSON is on ActivityStreams, in that case it could be just an ActivityStreams crate.

**Comment by asonix** [comment]
@stevenroose the [activitypub](https://docs.rs/activitypub) and [activitystreams](https://docs.rs/activitystreams) crates that I've built already do that.
